docs/userguide/active_learning.md

```diff
@@ -0,0 +1,66 @@
+# Active Learning
+When labeling unmeasured data points, e.g. for a data acquisition campaign to gather 
+data for a machine learning model, it can be beneficial to follow a guided approach 
+rather than randomly measuring data. If this is done via iteratively measuring points 
+according to a criterion proportional to the current model's uncertainty, the 
+method is called **active learning**.
+
+Active learning can be seen as a special case of Bayesian optimization. If we have the 
+above-mentioned criterion and set up a Bayesian optimization campaign to recommend 
+points with the highest uncertainty, we achieve active learning via Bayesian 
+optimization. In practice, we can leverage the fact that we have a posterior 
+distribution, and compute an acquisition function that represents the overall 
+uncertainty about any given point in the searchspace.
```

Comment:
I'm not yet super happy with this paragraph, perhaps you can do another iteration? What I don't like so much:
* The content is quite redundant to what was already mentioned before, and words repeat
* Don't know how to understand the "in practice". What has the content to do with "in practice"? I think this is rather the theoretic / conceptual background (i.e. you'd do exactly what you have described even if you conducted in only with pen and paper)

Reply:
tbh I dont get your points
this sentence is not redundant, it lays out how AL is a special case of BO, not mentioend anywhere before. This is followed dby how a GP calculates uncertainty in practice, which is the concrete implementation of the previously mentioned abstract concept, hence `in practice`
please be more specific if there is text that need changing or push a commit proposal

Reply:
Hm, I would actually see this from a different angle. "Uncertainty" is just layman's terms for the "spread" of the posterior distribution. So it's not that one starts with "uncertainty" and than uses a "posterior distribution" in practice as an instantiation of the uncertainty metric — I'd say it's the other way round: You start with the posterior distribution right away, that's how you define the optimization problem in the first place. Or are you saying there is a way to formally write down the problem without using distributions? To be clear: I'll let "pseudocode" alone using "find point that maximizes uncertainty" only count if you can give a concrete example of an algorithm that cannot be derived as a special case of a distribution-based view. 🙃

Reply:
- that is a very GP centered view, what if I use modelling that doesnt come with posteriors and still want to quantify uncvertainty? This is why imo its pretty obvious that a good explanation goes `uncertainty in general -> uncertainty in practice` a la `uncertainty for any model -> quantify uncertainty for a GP`
- Lets not treat  PR review as a paper review, it wastes time and certainly doesnt produce the best userguide

Reply:
Interesting that you keep bringing up the GP (I already was about to ask you in you previous post why you specifically mention it). In fact, I never explicitly mention GPs, my argument is valid for all kinds of probabilistic models...

Anyway, let's try to get forward. I've just pushed another commit with a slight rephrasing of the procedure. Let me know if you are happy with it. It also fixes the statement `acquisition function that represents the overall 
uncertainty about any given point in the searchspace` which is really not generally correct. For example, the `qNIPV` doesn't work at all like that. Perhaps we converge in this way 🙃

Reply:
all fine for me, still awaiting acceptance

Reply:
now you got it 🙃